e2e/vertical-full-page-map.spec.ts

```diff
@@ -0,0 +1,96 @@
+    const detachedSearchLogo = await page.$(searchLogo);
+    expect(detachedSearchLogo).toBeFalsy();
+  });
+
+  test('search this area button works', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    const responsePromise = await page.waitForResponse(/https:\/\/prod-cdn\.us\.yextapis\.com\/v2\/accounts\/me\/search\/vertical/i);
+    await page.getByLabel('Map controls').getByText('Search When Map Moves').click();
+    const response = responsePromise;
+  });
+
+  test('default initial search works and is enabled by default', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    const result = page.locator('#js-answersVerticalResults div').nth(0);
+    await expect(result).toBeAttached();
+  });
+
+  test('pagination works', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    await page.getByLabel('Go to the next page of results').click();
+    const secondPage = page.locator('#js-answersVerticalResultsCount');
+    await expect(secondPage).toHaveText(/21/);
+  });
+
+  test('pagination scrolls the results to the top', async ({ page }) => {
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+    await page.getByLabel('Go to the next page of results').click();
+    const locator = page.locator('#js-answersVerticalResults div').nth(0);
+    await expect(locator).toBeVisible();
+  });
+
+});
+
+test.describe('full page map with filters test suite', () => {
+  test.beforeEach(async ({page}) => {
+    await page.goto('http://localhost:5042/locations_full_page_map_with_filters');
+    await page.getByPlaceholder('Search for locations').fill('virginia');
+    await page.getByPlaceholder('Search for locations').press('Enter');
+  });
+
+  test('clicking on a pin closes the filter view', async ({ page }) => {
+    await page.getByRole('button', { name: 'filter results' }).click();
+    await page.getByText('Cats (1)').click();
+    const filterView = page.getByLabel('Main location search').locator('div').filter({ hasText: 'Filters Services reset Cats (1) Dogs (1) Sleep (1)' }).first();
+    await expect(filterView).toBeVisible();
+    await page.getByRole('button', { name: 'Result number 1' }).click();
+    await expect(filterView).not.toBeVisible();
+  });
+
+  test('clicking on a cluster causes the map to zoom in', async ({ page }) => {
+    const mapboxPinCount = await page.locator('#js-answersMap div').count();
+    await page.getByRole('button', { name: 'Cluster of 2 results' }).click();
+    const mapboxPinCountAfterSelectingCluster = await page.locator('#js-answersMap div').count();
+    expect(mapboxPinCount).not.toBe(mapboxPinCountAfterSelectingCluster);
```

Comment:
can we check that there are fewer pins, rather than an arbitrary different number?